aiidalab/app.py

```diff
@@ -258,6 +262,31 @@ def find_incompatibilities(self, version, python_bin=None):
     def is_compatible(self, version, python_bin=None):
         return not any(self.find_incompatibilities(version, python_bin))
 
+    def find_dependencies_to_install(self, version_to_install, python_bin=None):
+        """Return a list of dependencies (namedtuple Dependency)."""
+        if python_bin is None:
+            python_bin = sys.executable
+
+        if not version_to_install:
+            return []
+
+        unmatched_requirements = [
+            r[1] for r in self.find_incompatibilities(version_to_install, python_bin)
+        ]
+        packages = find_installed_packages(python_bin)
+
+        dependencies_to_install = []
+        for requirement in unmatched_requirements:
+            installed = None
+
+            for package in packages:
+                if requirement.name == package.name:
+                    installed = package
+
+            dependencies_to_install.append(Dependency(installed, requirement))
+
+        return dependencies_to_install
```

Comment:
```suggestion

        p_dict = {p.name:p for p in packages}
        r_dict = {r.name:r for r in requirements}
        intersection = r_dict.keys() & p_dict.keys()
        return [Dependency(p_dict[i], r_dict[i]) for i in intersection]

```

Reply:
done in [2c3eab0](https://github.com/aiidalab/aiidalab/pull/347/commits/2c3eab0ad6d006b96a76c40990a0685ead87c2e6)